File: app/src/test/java/ru/androidschool/rxjava/CreateExample.java
package ru.androidschool.rxjava;

import org.junit.Test;

import io.reactivex.rxjava3.core.Observable;
import io.reactivex.rxjava3.core.ObservableEmitter;
import io.reactivex.rxjava3.core.ObservableOnSubscribe;
import io.reactivex.rxjava3.core.Observer;
import io.reactivex.rxjava3.disposables.Disposable;

public class CreateExample {

    @Test
    public void createDemoSample() {

        Observable<String> source = Observable.create(new ObservableOnSubscribe<String>() {
            @Override
            public void subscribe(ObservableEmitter<String> emitter) {
                emitter.onNext("Java");
                emitter.onNext("Kotlin");
                emitter.onNext("Go");
                emitter.onNext("Cpp");
                emitter.onNext("Python");
                emitter.onComplete();
            }
        });

        source.subscribe(new Observer<String>() {
            Disposable d;

            @Override
            public void onSubscribe(Disposable d) {
                this.d = d;
                System.out.println("onSubscribe: " + Thread.currentThread().getName());
            }

            @Override
            public void onNext(String s) {
                System.out.println("onNext: " + s);
            }

            @Override
            public void onError(Throwable e) {
                System.out.println("onError: " + e);
            }

            @Override
            public void onComplete() {
                System.out.println("onComplete");
            }
        });
    }
}

File: app/src/test/java/ru/androidschool/rxjava/kotlin/create/CreateExample.kt
package ru.androidschool.rxjava.kotlin.create


import io.reactivex.rxjava3.core.Observable
import io.reactivex.rxjava3.core.ObservableOnSubscribe
import io.reactivex.rxjava3.core.Observer
import io.reactivex.rxjava3.disposables.Disposable
import org.junit.Test

class CreateExample {

    @Test
    fun createDemoSample() {

        val source = Observable.create(ObservableOnSubscribe<String> { emitter ->
            emitter.onNext("Java")
            emitter.onNext("Kotlin")
            emitter.onNext("Go")
            emitter.onNext("Cpp")
            emitter.onNext("Python")
            emitter.onComplete()
        })

        source.subscribe(object : Observer<String> {
            var d: Disposable? = null

            override fun onSubscribe(d: Disposable) {
                this.d = d
                println("onSubscribe: " + Thread.currentThread().name)
            }

            override fun onNext(s: String) {

                println("onNext: $s")
            }

            override fun onError(e: Throwable) {
                println("onError: $e")
            }

            override fun onComplete() {
                println("onComplete")
            }
        })
    }
}